Panic on `kots install` and `kots admin-console` if ports are in use

If you run `kots install` or `kots admin-console` while another such instance is running, kots exits with the following:

```
panic: Unable to listen on any of the requested ports: [{8800 3000}]

goroutine 12 [running]:
github.com/replicatedhq/kots/pkg/k8sutil.PortForward.func2(0xc0000f2840, 0xc00033a7d0, 0xc00025c7b0, 0xc00036bbb8)
	/go/src/github.com/replicatedhq/kots/pkg/k8sutil/portforward.go:63 +0x82
created by github.com/replicatedhq/kots/pkg/k8sutil.PortForward
	/go/src/github.com/replicatedhq/kots/pkg/k8sutil/portforward.go:61 +0x707
```

We could handle this better.

--- a/cmd/kots/cli/download.go
+++ b/cmd/kots/cli/download.go
@@ -15,7 +15,7 @@ func DownloadCmd() *cobra.Command {
 		Short:         "",
 		Long:          ``,
 		SilenceUsage:  true,
-		SilenceErrors: false,
+		SilenceErrors: true,
 		PreRun: func(cmd *cobra.Command, args []string) {
 			viper.BindPFlags(cmd.Flags())
 		},

--- a/pkg/download/download.go
+++ b/pkg/download/download.go
@@ -50,14 +50,14 @@ func Download(appSlug string, path string, downloadOptions DownloadOptions) erro
 
 	tmpFile, err := ioutil.TempFile("", "kots")
 	if err != nil {
-		log.FinishSpinnerWithError()
+		log.FinishSpinner()
 		return errors.Wrap(err, "failed to create temp file")
 	}
 	defer os.Remove(tmpFile.Name())
 
 	_, err = io.Copy(tmpFile, resp.Body)
 	if err != nil {
-		log.FinishSpinnerWithError()
+		log.FinishSpinner()
 		return errors.Wrap(err, "failed to write archive")
 	}
 
@@ -68,6 +68,8 @@ func Download(appSlug string, path string, downloadOptions DownloadOptions) erro
 				return errors.Wrap(err, "failed to delete existing download")
 			}
 		} else {
+			log.FinishSpinner()
+			log.Info("Directory %s already exists. You can re-run this command with --overwrite to automatically overwrite it", path)
 			return errors.Errorf("directory already exists at %s", path)
 		}
 	}

--- a/pkg/k8sutil/portforward.go
+++ b/pkg/k8sutil/portforward.go
@@ -3,16 +3,34 @@ package k8sutil
 import (
 	"bytes"
 	"fmt"
+	"net"
 	"net/http"
 	"net/url"
+	"strconv"
 	"time"
 
+	"github.com/pkg/errors"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/portforward"
 	"k8s.io/client-go/transport/spdy"
 )
 
+func IsPortAvailable(port int) bool {
+	host := ":" + strconv.Itoa(port)
+	server, err := net.Listen("tcp", host)
+	if err != nil {
+		return false
+	}
+
+	server.Close()
+	return true
+}
+
 func PortForward(kubeContext string, localPort int, remotePort int, namespace string, podName string) (chan struct{}, error) {
+	if !IsPortAvailable(localPort) {
+		return nil, errors.Errorf("Unable to connect to cluster. There's another process using port %d.", localPort)
+	}
+
 	// port forward
 	config, err := clientcmd.BuildConfigFromFlags("", kubeContext)
 	if err != nil {

